squire_lib/src/ffi_tournament.rs

```diff
@@ -86,6 +123,173 @@ impl TournamentId {
         }
     }
 
+    /// Updates the settings
+    /// Values that are not to be changed should remain the
+    /// current setting, that would be the value the user
+    /// selected in the GUI so that is fine.
+    /// All input must be non-null.
+    ///
+    /// If any errors occur then all actions are rolled back
+    /// and, false returned.
+    ///
+    /// Otherwise true is returned and the operations are all
+    /// applied to the tournament.
+    #[no_mangle]
+    pub extern "C" fn tid_update_settings(
+        self,
+        tid: TournamentId,
+        __format: *const c_char,
+        starting_table_number: u64,
+        use_table_number: bool,
+        game_size: u8,
+        min_deck_count: u8,
+        max_deck_count: u8,
+        reg_open: bool,
+        require_check_in: bool,
+        require_deck_reg: bool,
+    ) -> bool {
+        let mut tournament: &Tournament;
+        unsafe {
```

Comment:
Unnecessary unsafe?

Reply:
`.get_mut()` is unsafe.

Reply:
You shouldn't need `get_met()`. The `DashMap` handles mutual exclusion for us.

Reply:
resolved in c68cf6a3